analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/ProgramStateTest.cs

```diff
@@ -358,13 +351,12 @@ public void ToString_WithSymbols()
 a: SV_1
 ");
 
-            var valueWithConstraint = new SymbolicValue(counter);
-            valueWithConstraint.SetConstraint(TestConstraint.Second);
+            var valueWithConstraint = new SymbolicValue(counter).WithConstraint(TestConstraint.Second);
             sut = sut.SetSymbolValue(variableSymbol.ContainingSymbol, valueWithConstraint);
             sut.ToString().Should().BeIgnoringLineEndings(
 @"Symbols:
 a: SV_1
-Sample.Main(): SV_2: Second
+Sample.Main(): SV_3: Second
```

Comment:
Not sure I understand why is this 3. Do we explore more paths and create more symbolic values?

Reply:
No, the `With....` creates a new instance due to immutability and increases the counter in the constructor that looks like unused.

Reply:
Thanks for clarifying this.